test/e2e/grpc_client.go

```diff
@@ -0,0 +1,90 @@
+// Copyright 2022 The Tekton Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e
+
+import (
+	"context"
+	"crypto/x509"
+	"errors"
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	"google.golang.org/grpc/credentials/oauth"
+	"k8s.io/client-go/transport"
+
+	resultsv1alpha2 "github.com/tektoncd/results/proto/v1alpha2/results_go_proto"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+)
+
+const (
+	certFile                  = "/tmp/tekton-results-cert.pem"
+	apiServerName             = "tekton-results-api-service.tekton-pipelines.svc.cluster.local"
+	apiServerAddress          = "localhost:50051"
+	allNamespacesReadAccess   = "/tmp/tekton-results/tokens/all-namespaces-read-access"
+	singleNamespaceReadAccess = "/tmp/tekton-results/tokens/single-namespace-read-access"
```

Comment:
Likewise, when this is made configurable, we should move these constants near a test main.